services/proxy/pkg/middleware/oidc_auth.go

```diff
@@ -146,6 +206,59 @@ func (m oidcAuth) shouldServe(req *http.Request) bool {
 	return strings.HasPrefix(header, "Bearer ")
 }
 
+type jwksJSON struct {
+	JWKSURL string `json:"jwks_uri"`
+}
+
+func (m *oidcAuth) getKeyfunc() *keyfunc.JWKS {
+	if m.jwks == nil {
```

Comment:
Will this need some locking and/or syncing? Or are we sure we always run this single threaded?

Reply:
Hm, good point. I guess the same is true for the `getProvider` function below. Need to take a closer look.

Reply:
Fixed